pallets/parachain-system/src/migration.rs

```diff
@@ -14,39 +14,53 @@
 // You should have received a copy of the GNU General Public License
 // along with Cumulus.  If not, see <http://www.gnu.org/licenses/>.
 
-use crate::{Config, Pallet};
+use crate::{Config, Pallet, Store};
+use cumulus_primitives_core::relay_chain::v2::MAX_POV_SIZE;
 use frame_support::{
 	traits::{Get, StorageVersion},
 	weights::Weight,
 };
 
 /// The current storage version.
-pub const STORAGE_VERSION: StorageVersion = StorageVersion::new(1);
+pub const STORAGE_VERSION: StorageVersion = StorageVersion::new(2);
 
 /// Call this during the next runtime upgrade for this module.
 pub fn on_runtime_upgrade<T: Config>() -> Weight {
-	let mut weight: Weight = Weight::zero();
+	let mut weight: Weight = T::DbWeight::get().reads(1);
 
-	if StorageVersion::get::<Pallet<T>>() == 0 {
+	if StorageVersion::get::<Pallet<T>>() == 1 {
 		weight = weight
-			.saturating_add(v1::migrate::<T>())
+			.saturating_add(v2::migrate::<T>())
 			.saturating_add(T::DbWeight::get().writes(1));
 		StorageVersion::new(1).put::<Pallet<T>>();
 	}
 
 	weight
 }
 
-/// V1: `LastUpgrade` block number is removed from the storage since the upgrade
-/// mechanism now uses signals instead of block offsets.
-mod v1 {
```

Comment:
I'm unsure if V1 has already enacted on chain, perhaps I shouldn't have replaced the code here?